src/Network/Packet/Serverbound/Play/UseItemOnPacket.php

```diff
@@ -47,5 +50,58 @@ public function handle(Session $session): void
             ->getBlock($this->position);
 
         $server->broadcastPacket(new BlockUpdatePacket($this->position, $block));
+
+        $blockData = $block->getMaterial()->getKey();
+        $player = $session->getPlayer();
+        
+        $chestBoats =[
+            Material::OAK_CHEST_BOAT->getKey(),
+            Material::BIRCH_CHEST_BOAT->getKey(),
+            Material::DARK_OAK_CHEST_BOAT->getKey(),
+            Material::ACACIA_CHEST_BOAT->getKey(),
+            Material::BAMBOO_CHEST_RAFT->getKey(),
+            Material::JUNGLE_CHEST_BOAT->getKey(),
+            Material::SPRUCE_CHEST_BOAT->getKey(),
+            Material::MANGROVE_CHEST_BOAT->getKey(),
+            Material::PALE_OAK_CHEST_BOAT->getKey(),
+        ];
+
+        $blockList = [
+            Material ::CHEST_MINECART->getKey() => [2, 'Minecart with Chest'],
+            Material ::CHEST->getKey() => [2,'Chest'],
+            Material ::TRAPPED_CHEST->getKey() => [2,'Trapped Chest'],
+            Material ::ENDER_CHEST->getKey() => [2,'Ender Chest'],
+            Material ::BARREL->getKey() => [2,'Barrel'],
+            Material ::DISPENSER->getKey() => [6, 'Dispenser'],
+            Material ::DROPPER->getKey() => [6,'Dropper'],
+            Material ::CRAFTER->getKey() => [7, 'Crafter'],
+            Material ::ANVIL->getKey() => [8, 'Anvil'],
+            Material ::BEACON->getKey() => [9, 'Beacon'],
+            Material ::BLAST_FURNACE->getKey() => [10, 'Blast Furnace'],
+            Material ::BREWING_STAND->getKey() => [11, 'Brewing Stand'],
+            Material ::CRAFTING_TABLE->getKey() => [12, 'Crafting Table'],
+            Material ::ENCHANTING_TABLE->getKey() => [13, 'Enchanting Table'],
+            Material ::FURNACE->getKey() => [14, 'Furnace'],
+            Material ::GRINDSTONE->getKey() => [15, 'Grindstone'],
+            Material ::HOPPER->getKey() => [16, 'Hopper'],
+            Material ::HOPPER_MINECART->getKey() => [16, 'Minecart with Hopper'],
+            Material ::LECTERN->getKey() => [17, 'Lectern'],
+            Material ::LOOM->getKey() => [18, 'Loom'],
+            Material ::SHULKER_BOX->getKey() => [20, 'Shulker Box'],
+            Material ::SMITHING_TABLE->getKey() => [21, 'Smithing Table'],
+            Material ::SMOKER->getKey() => [22, 'Smoker'],
+            Material ::CARTOGRAPHY_TABLE->getKey() => [23, 'Cartography Table'],
+            Material ::STONECUTTER->getKey() => [24, 'Stonecutter'],
+        ];
```

Comment:
Create a class for each type of inventory; this will make it easier to manage the API.